src/Microsoft.Sbom.Api/Manifest/Configuration/SbomConfigProvider.cs

```diff
@@ -87,12 +88,16 @@ public class SbomConfigProvider : ISbomConfigProvider
         /// <inheritdoc/>
         public ISbomConfig Get(ManifestInfo manifestInfo)
         {
-            if (manifestInfo is null)
+            ArgumentNullException.ThrowIfNull(nameof(manifestInfo));
+
+            if (ConfigsDictionary.TryGetValue(manifestInfo, out var value))
             {
-                throw new ArgumentNullException(nameof(manifestInfo));
+                return value;
+            }
+            else
+            {
+                throw new FileNotFoundException($"Unable to handle a manifest of type {manifestInfo} or no manifest file was found.");
```

Comment:
Is it possible to check for the existence of the file at this point to distinguish the two different causes of error?

Reply:
I don't believe so.  At this point in the code, we are dealing with instances of `IManifestConfigHandler`, which only exposes a single method, `TryGetManifestConfig`.  

We could have the various implementations of `IManifestConfigHandler` throw a `FileNotFoundException` though.  I'm OK doing this instread, although we just need to ensure that each handler handles this condition.

At this point in the code, all we know is that there's no configuration available for the manifestInfo in question.  We can only infer that 1) no handler is available or 2) that no manifest is present.